### PR TITLE
[ide] chop sentences taking into account QUOTATION token

### PR DESCRIPTION
**Pull request by gares**
This is the shamefully forgotten part of #9733

### CODE REVIEW

**Comment by gares** [comment]
Without this PR, #9733 works fine with coqc but not with coqide...

**Comment by gares** [comment]
whitespace errors fixed, liter can RIP

**Comment by Zimmi48** [comment]
The failure seems unrelated. Sync issue?

**Comment by gares** [comment]
I think so, but I've just restarted dpdgraph in case it gets updated

**Review by ppedrot — APPROVED** [review]
Forgot to mention this looked OK.

**Review by ppedrot — CHANGES_REQUESTED** [review]
So, actually this is breaking legitimate use cases. The following code doesn't parse anymore in CoqIDE after this patch.
```
Goal True.
Proof.
let t := constr:(True) in idtac.
```

**Comment by gares** [comment]
Well, it is supposed to be accepted by the lexer, so there is a bug lurking. I did interactive tests for stuff like `foo:(bla. bla).` in order to check that the middle `.` was not mistaken for an end of sentence. I'll try to amend ASAP.

Unfortunately it is not easy to test the lexer in CI :-/

**Comment by SkySkimmer** [comment]
>Unfortunately it is not easy to test the lexer in CI :-/

It can't be called in a unit test?

**Comment by gares** [comment]
It now works for the example above, I'll give a try to the unit test business tomorrow

**Comment by gares** [comment]
We now have some tests using the unit-test framework, this should avoid re-introducing the mistake found by @ppedrot 
I had to fix the META file wrt coq.ide (@ejgallego) and to add a few lines to the test-suite/Makefile thingy (@SkySkimmer )

**Comment by gares** [comment]
linter RIP

File: test-suite/unit-tests/ide/lex_tests.ml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+open Utest
```
Comment (SkySkimmer) [inline test-suite/unit-tests/ide/lex_tests.ml]: Do we have a mechanism to skip ide tests when gtk isn't available?
Alternatively, coq_lex depends on gtk to pass `Tags.Script.comment` and `Tags.Script.sentence` to the closure it takes as argument, maybe we could refactor to remove the dependency and move coq_lex to ide_common.

Comment (gares) [reply test-suite/unit-tests/ide/lex_tests.ml]: Is there a CI setup running test-suite and not having lablgtk installed?

Comment (gares) [reply test-suite/unit-tests/ide/lex_tests.ml]: Apparently no, so this PR seems OK. But I see your point, and my preferred solution would be making the lexer not depend on GTK ;-)

**Comment by gares** [comment]
@ppedrot would you mind updating your review?

**Review by ppedrot — APPROVED** [review]
I wasn't able to break it, which doesn't mean it's correct, but at least it looks so.

**Comment by gares** [comment]
If you give me your tests, I can put them here: https://github.com/coq/coq/pull/10394/files#diff-f5f6768b60643e0d2b5c56790038b49aR23

**Comment by ppedrot** [comment]
What about the one given above that was broken in this thread?

(I didn't keep my session, just tried haphazardly to mess with quotation-looking sentences.)

**Comment by gares** [comment]
That one is part of the test-suite

**Comment by gares** [comment]
Truth is that UIs should stop to parse, but LSP is not there yet :-/

I hope the whole coq_lex will go one day...

**Comment by gares** [comment]
ping

**Comment by ppedrot** [comment]
Let's merge, CI failure is spurious.

**Pull request merged**